lib/small_url_web/live/stats_live.ex

```diff
@@ -1,56 +1,107 @@
 
-    if connected?(socket), do: SmallUrlWeb.UrlController.subscribe()
-    
-    link = Links.get_short_links_by_key(key)
+    shortlink = Links.get_short_links_by_key(key)
 
-    shortlink = SmallUrl.Repo.preload(link, :clicks)
+    shortlink = SmallUrl.Repo.preload(shortlink, :clicks)
     clicks = Map.get(shortlink, :clicks)
     number_of_clicks = Enum.count(clicks)
 
     last_click_date =
       Enum.map(clicks, fn click -> Map.get(click, :click_date) end) |> Enum.sort() |> Enum.at(0)
 
-    clicks_from_last_30_days =
-      Links.clicks_from_last_30_days(shortlink.key)
-      |> Enum.map(fn click -> Map.get(click, :click_date) end)
+    click_frequencies = get_click_frequencies_for_last_30_days(key)
 
     link_info = %{
       :clicks => number_of_clicks,
       :last_click_date => last_click_date,
-      :last_30_days => clicks_from_last_30_days
+      :last_30_days => 0
+    }
+
+    {:ok,
+     assign(socket, %{
+       link: shortlink,
+       link_info: link_info,
+       key: key,
+       click_frequencies: click_frequencies
+     })}
+  end
+
+  def handle_info(:update_click_frequency_graph, socket) do
+    key = socket.assigns[:key]
+
+    shortlink = Links.get_short_links_by_key(key)
+
+    click_frequencies = get_click_frequencies_for_last_30_days(key)
+
+    shortlink = SmallUrl.Repo.preload(shortlink, :clicks)
+    clicks = Map.get(shortlink, :clicks)
+    number_of_clicks = Enum.count(clicks)
+
+    link_info = %{
+      :clicks => number_of_clicks,
+      :last_click_date => nil
     }
 
-    {:ok, assign(socket, %{link: link, link_info: link_info, key: key})}
+    {:noreply,
+     assign(socket,
+       link_info: link_info,
+       click_frequencies: click_frequencies
+     )}
   end
 
   @impl true
   def handle_info(map, socket) do
     key = socket.assigns[:key]
 
-    link = Links.get_short_links_by_key(key)
+    shortlink = Links.get_short_links_by_key(key)
 
-    shortlink = SmallUrl.Repo.preload(link, :clicks)
+    shortlink = SmallUrl.Repo.preload(shortlink, :clicks)
     clicks = Map.get(shortlink, :clicks)
     number_of_clicks = Enum.count(clicks)
 
     last_click_date =
       Enum.map(clicks, fn click -> Map.get(click, :click_date) end) |> Enum.sort() |> Enum.at(0)
 
-    clicks_from_last_30_days =
-      Links.clicks_from_last_30_days(shortlink.key)
-      |> Enum.map(fn click -> Map.get(click, :click_date) end)
-
     link_info = %{
       :clicks => number_of_clicks,
       :last_click_date => last_click_date,
-      :last_30_days => clicks_from_last_30_days
+      :last_30_days => 0
     }
 
-    {:noreply, assign(socket, %{link: link, link_info: link_info})}
+    click_frequencies = get_click_frequencies_for_last_30_days(key)
+
+    {:ok,
+     assign(socket, %{
+       link: shortlink,
+       link_info: link_info,
+       key: key,
+       click_frequencies: click_frequencies
+     })}
+  end
+
+  defp get_click_frequencies_for_last_30_days(nil), do: [%{}]
```

Comment:
This can be removed, because the data it returns does not conform to the shape expected by the `VegaLite.from_values` function.